benchmarks/tosi_et_al_2015_gcubed/tosi.cc

```diff
@@ -149,19 +149,18 @@ namespace aspect
                       deta[0][0] = eta_zero_zero - out.viscosities[i];
 
                       if (dstrain_rate[0][0] != 0)
-                        deta[0][0] /= std::fabs(dstrain_rate[i][0][0]) * finite_difference_accuracy;
+                        deta[0][0] /= std::fabs(in.strain_rate[i][0][0]) * finite_difference_accuracy;
                       else
                         deta[0][0] = 0;
 
                       // derivative in xy direction
                       dstrain_rate = in.strain_rate[i];
-                      // 0.5 because symmetry doubles this entry
-                      dstrain_rate[1][0]  += 0.5 * std::fabs(in.strain_rate[i][1][0]) * finite_difference_accuracy; 
+                      dstrain_rate[1][0]  += std::fabs(in.strain_rate[i][1][0]) * finite_difference_accuracy; 
```

Comment:
unfortunately, this is not correct. I don't remember the details of why it is happening (long time ago and it is late now), but I remember that this needs to be there. Just to be sure I quickly ran a test to compare it against the analytic solution (just removing the if statement so that they both run and output both to screen), assuming the analytic solution is correct of course ;)

Without 0.5: 
```
analytical derivative = 1.43213e-10 -1.33969e-12 -1.33969e-12 -1.44939e-10
finite difference derivative = 1.43212e-10 -2.76208e-12 -2.76208e-12 -1.4494e-10
```
with the 0.5:
```
analytical derivative = 1.43213e-10 -1.33969e-12 -1.33969e-12 -1.44939e-10
finite difference derivative = 1.43212e-10 -1.38104e-12 -1.38104e-12 -1.4494e-10
```

I also tested the change from dstrain_rate to in.strain_rate and that didn't matter at all:
```
analytical derivative = 1.43213e-10 -1.33969e-12 -1.33969e-12 -1.44939e-10
finite difference derivative = 1.43212e-10 -1.38104e-12 -1.38104e-12 -1.4494e-10
```

Reply:
Thanks for checking, I think I figured out the reason. When we modify the yx (1,0) component of the strain rate we implicitly also modify the xy (0,1) component, because it is a symmetric tensor. Therefore the delta in strain rate when computing the derivative for the off-diagonal terms is actually bigger than the ` 1e-7 * value` unless we account for the symmetry. I removed the change and added documentation.

> I also tested the change from dstrain_rate to in.strain_rate and that didn't matter at all:

Well it may be too small to make a difference here, but do you agree that we need to divide by the actual delta we modify the strain rate by, and not the modified value of the strain rate times the relative difference? The difference is likely around 1e-7 times the derivative.

Reply:
Ah, yes, that makes sense. We can't check them independently due to symmetry. Thanks for expanding the documentation!

I was already a bit worried when I looked again at the viscoplastic derivatives, but this makes me more worried, so I will need to take a closer look at that. (worried in the sense that it is not optimal. If that is the case we might actually improve the convergence rate)

> Well it may be too small to make a difference here, but do you agree that we need to divide by the actual delta we modify the strain rate by, and not the modified value of the strain rate times the relative difference? The difference is likely around 1e-7 times the derivative.

Yes, I agree that that is more correct.